lib/cdo/shared_constants/curriculum/shared_course_constants.rb

```diff
@@ -204,5 +204,13 @@ module SharedCourseConstants
         facilitator: ['student', 'teacher', 'facilitator']
       }
     ).freeze
+
+    NUMBERED_UNITS_TYPE = OpenStruct.new(
+      {
+        none: "none",
```

Comment:
this is a bit fuzzy as a shared constant -- it's an accurate list of valid values for js, but not fully accurate for ruby because it seems we never have a value of "none" in ruby. 

I think it could be worth locking this down so that there's less room for getting into a bad state. here is one idea:
- turn numbered_units into its own column, non null, with default value "none"
- use a validation to ensure correct value, like we do here: https://github.com/code-dot-org/code-dot-org/blob/6bce6d43717171dba30d0bd74208637ba72c216f/dashboard/app/models/unit_group.rb#L69

another idea would be to just use `nil` as one of the valid values instead of "none", which should get safely translated into `null` in shared constants js.